File: connectx-backend/src/main/java/com/connectx/group/dto/CreateGroupInvitationRequestDto.java
package com.connectx.group.dto;

import jakarta.validation.constraints.NotNull;

/**
 * Deliberately carries only the target user id -- no inviterId, no role, no membershipStatus.
 * The inviter is always the authenticated principal (see GroupInvitationController) and every
 * authorization question (permission, blocking, capacity, target state/privacy/connection) is
 * decided server-side by GroupAuthorizationService, never influenced by anything in this DTO.
 */
public class CreateGroupInvitationRequestDto {

    @NotNull(message = "targetUserId is required")
    private Long targetUserId;

    public CreateGroupInvitationRequestDto() {}

    public CreateGroupInvitationRequestDto(Long targetUserId) {
        this.targetUserId = targetUserId;
    }

    public Long getTargetUserId() {
        return targetUserId;
    }

    public void setTargetUserId(Long targetUserId) {
        this.targetUserId = targetUserId;
    }
}

File: connectx-android/app/src/main/java/com/connectx/app/data/remote/group/model/CreateGroupInvitationRequestDto.kt
package com.connectx.app.data.remote.group.model

import kotlinx.serialization.Serializable

/**
 * Mirrors com.connectx.group.dto.CreateGroupInvitationRequestDto exactly.
 * Deliberately carries only the target user id -- no inviterId, role, or status;
 * every authorization question (permission, blocking, capacity, connection) is
 * decided server-side. Backend @NotNull -- required.
 */
@Serializable
data class CreateGroupInvitationRequestDto(
    val targetUserId: Long
)
